internal/controller/api_to_config.go

```diff
@@ -166,11 +178,20 @@ func neighborToFRR(n v1beta1.Neighbor, prefixesInRouter []string, alwaysBlock []
 		return nil, fmt.Errorf("neighbor %s has invalid DynamicASN %s specified, must be one of %s,%s", neighborName(n), n.DynamicASN, v1beta1.InternalASNMode, v1beta1.ExternalASNMode)
 	}
 
+	address := n.Address
+	unnumbered := false
+	if n.Interface != "" {
+		unnumbered = true
+		address = n.Interface
+		neighborFamily = ipfamily.DualStack
+	}
+
 	res := &frr.NeighborConfig{
 		Name:            neighborName(n),
 		ASN:             asnFor(n),
 		SrcAddr:         n.SourceAddress,
-		Addr:            n.Address,
+		Addr:            address,
```

Comment:
I don't like the fact that we hide the interface under Addr here. Can we have an explicit field?

Reply:
I agree that name Addr and the value looks not ideal.

Given that FRR itself is using the word PEER to indicate either Addr or Interface name  `neighbor PEER interface remote-as <internal|external|auto|ASN` 

What do  you think  about renaming  .Addr to .Peer? Or you prefer to have .Interface field?

I am not in favor of two field because it means everytime we are using .Addr (e.g. error) we should add logic if .Addr empty use the .Interface (but is not that we currently reference that a lot, is more about the future, so not strong argument)

(https://docs.frrouting.org/en/latest/bgp.html#clicmd-neighbor-PEER-interface-remote-as-internal-external-auto-ASN)

Reply:
I would've agreed if we hadn't to make logic on the fact that it is unnumbered or not, but we are doing it:

```
neighbor {{.neighbor.Addr}}{{ if .neighbor.Unnumbered }} interface{{ end }} remote-as {{.neighbor.ASN}}
```

if that's the case I'd prefer having a clear api where fields named explicitly and if we need to call out errors, have a function that returns the proper string (as we do elsewhere).
Names and fields are (almost) free, and we should strive for clarity first.

Reply:
not done, TODO